### DISCUSSION TITLE
How can I update the values of a ui.select element

### DISCUSSION THREAD

**Discussion by cschmidtoptima** [zauberzeug/nicegui#1405] [Q&A] [answered]
### Question

I have a ui with tabs layout. In one tab I add values to the database, and in the other tab i generate a select with values from this database. However I dont understand how i can trigger the update of the select field when someone adds values to the database. For now it works only when I restart the nicegui app.

Here is some code:

```py
       def get_customers_for_addfw():
        
            tuples_list = []
            query = select(customers.c.no,customers.c.name) 
            ResultProxy = connection.execute(query)
            ResultSet = ResultProxy.fetchall()
            for cus in ResultSet:
                    #print(cus)
                    tuples_list.append(list(cus))                  
            #print(tuples_list)
            my_dict = {t[0]: t[1] for t in tuples_list}
                                 #print(dict(cus._mapping))
    
     
            return my_dict

@ui.page('/')
async def page_layout() -> None:
 with ui.tab_panel('afw'):
    with ui.dialog() as dialog,ui.card():
                 await gen_fw_add_ui()

    @ui.refreshable     
    async def gen_fw_add_ui():
                    with ui.row():
                       
                        with ui.row():
               
                            
                    with ui.column():
                        ...
                        get_customers_result = get_customers_for_addfw()
                        fw_customer = ui.select( get_customers_result, with_input=True,value="Customer").props('outlined dense').props("size=50")
 ...
                   
```             

As this is quite a basic thing to know when you want to make a dynamic page, I would really like to understand how it works

**Comment by rodja** [comment, marked as answer]
You need either a way to get notified about a data change, or regularly check for modifications with a `ui.timer`. The update itself can then be done by modifying the `.options` property of your `ui.select` element. Do not forget to call `.update()` to let the modifications take effect. I've created pull request https://github.com/zauberzeug/nicegui/pull/1409 improve on that in the next version.

**Reply by falkoschindler** [reply]
PR https://github.com/zauberzeug/nicegui/pull/1409 is merged and will be released shortly.
